File: Firebase/creative_api/functions/src/Service/SupplyService.js
const Supply = require("./../Class/Supply");
const supplyRepository = require("./../Repository/SupplyRepository");

class SupplySevice {
    async saveSupply(supply) {
        return (await supplyRepository.save(supply.get()));
    }
    async getById(id) {
        return (await supplyRepository.find(id));
    }
    async getAll() {
        return (await supplyRepository.findAll());
    }
    async updateSupply(supply){
        return (await supplyRepository.update(supply.get()));
    }
}

module.exports = new SupplySevice();

File: C#/Truckleer/Truckleer.Creative/Modules/Supply/SupplyService.cs

using System.Collections.Generic;

namespace Truckleer.Modules
{
    public class SupplyService
    {
        //Supply repository property
        readonly SupplyRepository repository;
        //Class constructor
        public SupplyService()
        {
            //initializate supply repository property
            repository = new SupplyRepository();
        }
        //Method to findAll supplys
        public List<Supply> FindAll()
        {
            return repository.FindAll().Result;
        }
        //Method to find one specific supply by id
        public Supply FindOne(string Id)
        {
            return repository.Find(Id).Result;
        }
        //Method to Save or Update supply
        public bool Save(Supply supply)
        {
            return repository.Save(supply).Result;
        }
    }
}

